field/binary.go

```diff
@@ -129,12 +134,12 @@ func (f *Binary) Unmarshal(v interface{}) error {
 	return nil
 }
 
-func (f *Binary) SetData(data interface{}) error {
-	if data == nil {
+func (f *Binary) Marshal(v interface{}) error {
+	if v == nil {
 		return nil
 	}
 
-	bin, ok := data.(*Binary)
+	bin, ok := v.(*Binary)
```

Comment:
@mfdeveloper508 here and in the rest of the fields, we don't need `f.data` anymore. Before, it was used in `SetData` and was a pointer that was used when we set and get "data". With Marhsal and Unmarshl - we don't need to store the pointer to the field type as we use field value directly.

```go
        f.data = bin // <- the field can be removed
	if bin.value != nil {
		f.value = bin.value
	}
```

Reply:
https://github.com/moov-io/iso8583/pull/289
right fixed in next pr